### ISSUE TITLE
About potential error in bootstrap return computation

### ISSUE THREAD

**Issue by swsychen** [danijar/dreamerv3#190]
Hi, 

I'm referring to the old dreamerv3 version in commit "8fa35f8". In the bootstrap return computation within critic, the following lines (https://github.com/danijar/dreamerv3/blob/8fa35f83eee1ce7e10f3dee0b766587d0a713a60/dreamerv3/agent.py#L359C3-L371C32) seem not match the one in the paper:

```
def score(self, traj, actor=None):
    rew = self.rewfn(traj)          # rew from 1,....T since rewfn removed the first index 0 by " rewfn = lambda s: wm.heads['reward'](s).mean()[1:]"
    assert len(rew) == len(traj['action']) - 1, (
        'should provide rewards for all but last action')
    discount = 1 - 1 / self.config.horizon
    disc = traj['cont'][1:] * discount    # disc from 1,...T
    value = self.net(traj).mean()
    vals = [value[-1]]
    interm = rew + disc * value[1:] * (1 - self.config.return_lambda)
    for t in reversed(range(len(disc))):
      vals.append(interm[t] + disc[t] * self.config.return_lambda * vals[-1])
    ret = jnp.stack(list(reversed(vals))[:-1])
    return rew, ret, value[:-1]
```

**Problem:**
In this case, the bootstrap return of timestep T-1 still uses the reward at timestep T, since the last index in disc and interm points to timestep T. Making it basically: `R_{T-1} = r_T+ gamma*c_T*((1-lambda)*V_T + lambda*V_T)`. However, it should use the reward at timestep T-1 as shown in the paper:

<img width="380" height="80" alt="Image" src="https://github.com/user-attachments/assets/3870bcf4-328c-4a13-976d-ce16db7c961b" />

**Potential Fix:**
If we change the rewfn, to only remove the last index, then it should be correct. And the ` traj['cont'][1:]` should also be ` traj['cont'][:-1]`

Could you please let me know if I misunderstood something?

**Comment by danijar** [comment]
The environment returns `(r_t-1, d_t-1, o_t)` at the same time, and the world model is trained to do the same.
